### DISCUSSION TITLE
Question about CSR

### DISCUSSION THREAD

**Discussion by laxmisubramanyam** [ocelot-collab/ocelot#236] [General]
Hi, 

Sorry for bothering you again. But I reached to a point where test of CSR becomes essential and I am looking for some clarification about the effecient use of CSR. I have noticed that two methods are used in the examples of using CSR.

The first method is (used in Tutorial 5)
```
csr = CSR(n_bin=300, m_bin=5, sigma_min=0.2e-6)
```
and the second method is (used in csr_ex.py)
```
car = CSR()
csr.traj_step = 0.0002
csr.apply_step = 0.0002
```

What is the different between these two methods ? Is there any documentation about these two approaches ?

Also, what are the arguments for the default CSR() ?

Thank you.

Best regards,
Laxmi

**Comment by laxmisubramanyam** [comment]
Hi @ZeugAusHH, 

thanks for brining it to the discussion forum ( may be discussion will accelerate here !! ).

I made simulation of electron beam compressor using chicane. Later, I tried to see how CSR affects the electron beam phase-space. However, I didn't see any noticeable difference **with and without** CSR. This is strange to me. I would like to know if I am including the CSR correctly or not. 

Any help will be very appreciable. 

Thanks

**Comment by sergey-tomin** [comment]
Hi,
I would suggest to read doc string in CSR class https://github.com/ocelot-collab/ocelot/blob/master/ocelot/cpbd/csr.py 
if you will still have questions please ask we will try to help. 

Cheers,
Sergey.

**Comment by laxmisubramanyam** [comment]
Hi @sergey-tomin ,

Thanks a lot. I will follow your advice. 

Best regards,
Laxmi
